stdlib/dynarray.ml

```diff
@@ -962,6 +962,42 @@ let for_all p a =
   check_same_length "for_all" a ~length;
   res
 
+let exists2 p a1 a2 =
+  let Pack {arr = arr1; length = length1; dummy = dummy1} = a1 in
+  let Pack {arr = arr2; length = length2; dummy = dummy2} = a2 in
+  check_valid_length length1 arr1;
+  check_valid_length length2 arr2;
+  if length1 <> length2 then invalid_arg "Dynarray.exists2";
+  let rec loop p arr1 dummy1 arr2 dummy2 i length =
+    if i = length then false
+    else
+      p (unsafe_get arr1 ~dummy:dummy1 ~i ~length)
+        (unsafe_get arr2 ~dummy:dummy2 ~i ~length)
+      || loop p arr1 dummy1 arr2 dummy2 (i + 1) length
+  in
+  let res = loop p arr1 dummy1 arr2 dummy2 0 length1 in
+  check_same_length "exists2" a1 ~length:length1;
+  check_same_length "exists2" a2 ~length:length2;
+  res
+
+let for_all2 p a1 a2 =
+  let Pack {arr = arr1; length = length1; dummy = dummy1} = a1 in
+  let Pack {arr = arr2; length = length2; dummy = dummy2} = a2 in
+  check_valid_length length1 arr1;
+  check_valid_length length2 arr2;
+  if length1 <> length2 then invalid_arg "Dynarray.for_all2";
```

Comment:
`Dynarray` is the first indexed OCaml that structure that became programmer friendly in its error messages. Keep the spirit alive! 

There's an `Error` module above in which you should define a nice error message that indicates e.g.: `Dynarray.for_all2: array length mismatch: %d <> %d`.

Reply:
Great, fixed.